src/d/actor/d_a_majuu_flag.cpp

```diff
@@ -6,55 +6,794 @@
+        
+        if (iVar2 == 3) {
+            GXSetTevSwapMode(GX_TEVSTAGE2,GX_TEV_SWAP0,GX_TEV_SWAP2);
+            GXSetTevOrder(GX_TEVSTAGE2,GX_TEXCOORD1,GX_TEXMAP1,GX_COLOR_NULL);
+            GXSetTevColorIn(GX_TEVSTAGE2,GX_CC_ZERO,GX_CC_C2,GX_CC_TEXC,GX_CC_CPREV);
+            GXSetTevColorOp(GX_TEVSTAGE2,GX_TEV_ADD,GX_TB_ZERO,GX_CS_SCALE_1,GX_TRUE,GX_TEVPREV);
+            GXSetTevAlphaIn(GX_TEVSTAGE2,GX_CA_ZERO,GX_CA_ZERO,GX_CA_ZERO,GX_CA_TEXA);
+            GXSetTevAlphaOp(GX_TEVSTAGE2,GX_TEV_ADD,GX_TB_ZERO,GX_CS_SCALE_1,GX_TRUE,GX_TEVPREV);
+        }
+    } else {
+        GXSetChanCtrl(GX_COLOR0, GX_TRUE, GX_SRC_REG,GX_SRC_REG, lightMask, GX_DF_CLAMP,GX_AF_NONE);
+        GXSetNumTexGens(1);
+        GXSetTexCoordGen(GX_TEXCOORD0, GX_TG_MTX2x4, GX_TG_TEX0, GX_IDENTITY);
+        GXSetNumTevStages(iVar2);
+        GXSetTevOrder(GX_TEVSTAGE0,GX_TEXCOORD_NULL,GX_TEXMAP_NULL,GX_COLOR0A0);
+        GXSetTevSwapMode(GX_TEVSTAGE0,GX_TEV_SWAP1,GX_TEV_SWAP0);
+        GXSetTevColorIn(GX_TEVSTAGE0,GX_CC_C0,GX_CC_C1,GX_CC_RASC,GX_CC_ZERO);
+        GXSetTevColorOp(GX_TEVSTAGE0,GX_TEV_ADD,GX_TB_ZERO,GX_CS_SCALE_1,GX_TRUE,GX_TEVPREV);
+        GXSetTevAlphaIn(GX_TEVSTAGE0,GX_CA_ZERO,GX_CA_ZERO,GX_CA_ZERO,GX_CA_ZERO);
+        GXSetTevAlphaOp(GX_TEVSTAGE0,GX_TEV_ADD,GX_TB_ZERO,GX_CS_SCALE_1,GX_TRUE,GX_TEVPREV);
+        GXSetTevSwapMode(GX_TEVSTAGE1,GX_TEV_SWAP0,GX_TEV_SWAP0);
+        GXSetTevOrder(GX_TEVSTAGE1,GX_TEXCOORD0,GX_TEXMAP0,GX_COLOR_NULL);
+        GXSetTevColorIn(GX_TEVSTAGE1,GX_CC_ZERO,GX_CC_CPREV,GX_CC_TEXC,GX_CC_ZERO);
+        GXSetTevColorOp(GX_TEVSTAGE1,GX_TEV_ADD,GX_TB_ZERO,GX_CS_SCALE_1,GX_TRUE,GX_TEVPREV);
+        GXSetTevAlphaIn(GX_TEVSTAGE1,GX_CA_ZERO,GX_CA_ZERO,GX_CA_ZERO,GX_CA_TEXA);
+        GXSetTevAlphaOp(GX_TEVSTAGE1,GX_TEV_ADD,GX_TB_ZERO,GX_CS_SCALE_1,GX_TRUE,GX_TEVPREV);
+        
+        if (iVar2 == 3) {
+            GXSetTevSwapMode(GX_TEVSTAGE2,GX_TEV_SWAP2,GX_TEV_SWAP0);
+            GXSetTevOrder(GX_TEVSTAGE2,GX_TEXCOORD_NULL,GX_TEXMAP_NULL,GX_COLOR_NULL);
+            GXSetTevColorIn(GX_TEVSTAGE2,GX_CC_ZERO,GX_CC_C2,GX_CC_RASC,GX_CC_CPREV);
+            GXSetTevColorOp(GX_TEVSTAGE2,GX_TEV_ADD,GX_TB_ZERO,GX_CS_SCALE_1,GX_TRUE,GX_TEVPREV);
+            GXSetTevAlphaIn(GX_TEVSTAGE2,GX_CA_ZERO,GX_CA_ZERO,GX_CA_ZERO,GX_CA_TEXA);
+            GXSetTevAlphaOp(GX_TEVSTAGE2,GX_TEV_ADD,GX_TB_ZERO,GX_CS_SCALE_1,GX_TRUE,GX_TEVPREV);
+        }
+    }
+
+    GXSetTevColorS10(GX_TEVREG0, mpTevStr->mColorC0);
+    GXSetTevColor(GX_TEVREG1, mpTevStr->mColorK0);
+    GXSetTevColor(GX_TEVREG2, mpTevStr->mColorK1);
+    GXCallDisplayList(l_majuu_flag_matDL, 0x20);
+    GXLoadPosMtxImm(*getMtx(), 0);
+    GXLoadNrmMtxImm(*getMtx(), 0);
+    GXSetCullMode(GX_CULL_BACK);
+    GXCallDisplayList(l_majuu_flagDL, 0x80);
+    GXSetCullMode(GX_CULL_FRONT);
+#if VERSION == VERSION_DEMO
+    GXSetArray(GX_VA_NRM, mpNrmArrBack[mCurArr], sizeof(cXyz) * 1);
+#else
+    GXSetArray(GX_VA_NRM, getNrmBack(), sizeof(cXyz) * 1);
+#endif
```

Comment:
Where did `getNrmBack` come from? This inline doesn't seem to be in the debug map, but this also doesn't look like a genuine version difference between demo and retail since it's functionally equivalent, so I don't think there should be an ifdef here.

Reply:
I think it's a fakematch. Seems better to leave the regalloc and Nonmatching comment so that someone can try to figure out the real solution later.